packages/connection-storage/src/connection-storage.ts

```diff
@@ -1,520 +1,170 @@
+  }): Promise<ConnectionInfo[]>;
 
-  private static decryptSecrets(
-    secrets?: string
-  ): ConnectionSecrets | undefined {
-    return secrets
-      ? JSON.parse(safeStorage.decryptString(Buffer.from(secrets, 'base64')))
-      : undefined;
-  }
+  exportConnections(args?: {
+    options?: ExportConnectionOptions;
+    signal?: AbortSignal;
+  }): Promise<string>;
 
-  private static migrateSavedConnectionType(
-    connectionInfo: ConnectionInfo
-  ): ConnectionInfo {
-    const inferedConnectionType = connectionInfo.favorite
-      ? 'favorite'
-      : 'recent';
+  importConnections(args: {
+    content: string;
+    options?: ImportConnectionOptions;
+    signal?: AbortSignal;
+  }): Promise<void>;
+}
 
-    return {
-      ...connectionInfo,
-      savedConnectionType: connectionInfo.savedConnectionType
-        ? connectionInfo.savedConnectionType
-        : inferedConnectionType,
-    };
-  }
+export type CompassConnectionStorageIPCInterface = Omit<
+  CompassConnectionStorage,
+  'on' | 'off' | 'emit'
+>;
 
-  private static mapStoredConnectionToConnectionInfo({
-    connectionInfo: storedConnectionInfo,
-    connectionSecrets: storedConnectionSecrets,
-  }: ConnectionProps): ConnectionInfo {
-    let secrets: ConnectionSecrets | undefined = undefined;
-    try {
-      secrets = this.decryptSecrets(storedConnectionSecrets);
-    } catch (e) {
-      log.error(
-        mongoLogId(1_001_000_208),
-        'Connection Storage',
-        'Failed to decrypt secrets',
-        { message: (e as Error).message }
-      );
-    }
-    let connectionInfo = mergeSecrets(storedConnectionInfo!, secrets);
-    connectionInfo = this.migrateSavedConnectionType(connectionInfo);
-    return deleteCompassAppNameParam(connectionInfo);
-  }
+export type CompassConnectionStorageIPCMain = Pick<
+  HadronIpcMain,
+  'createHandle'
+>;
 
-  // This method is only called when compass tries to migrate connections to a new version.
-  // In e2e-tests we do not migrate any connections as all the connections are created
-  // in the new format, so keychain is not triggered (using keytar) and hence test is not blocked.
-  private static async getKeytarCredentials() {
-    try {
-      const credentials = await keytar.findCredentials(getKeytarServiceName());
-      return Object.fromEntries(
-        credentials.map(({ account, password }) => [
-          account,
-          parseStoredPassword(password),
-        ])
-      ) as Record<string, ConnectionSecrets>;
-    } catch (e) {
-      log.error(
-        mongoLogId(1_001_000_201),
-        'Connection Storage',
-        'Failed to load secrets from keychain',
-        { message: (e as Error).message }
-      );
-      return {};
-    }
-  }
+export type CompassConnectionStorageIPCRenderer = Pick<
+  HadronIpcRenderer,
+  'createInvoke'
+>;
 
-  private static async getConnections(): Promise<ConnectionWithLegacyProps[]> {
-    return (await this.userData.readAll()).data;
+export class NoopConnectionStorage
```

Comment:
Don't have a strong preference here, but if you implement this as some sort of in memory connection storage with maybe an option to pass initial state, you'd probably save yourself a lot of sinon stub code in tests 😄

Reply:
Yea this was indeed a life saviour. I ended up having an InMemoryConnectionStorage and that helped a lot with the tests